kata_agent.go

```diff
@@ -474,9 +476,13 @@ func (k *kataAgent) createContainer(pod *Pod, c *Container) (*Process, error) {
 		}
 
 		rootfs.Source = filepath.Join(devPath, driveName)
-		rootfs.MountPoint = rootPath // Should we remove the "rootfs" suffix?
+		rootfs.MountPoint = rootPathParent
 		rootfs.Fstype = c.state.Fstype
 
+		if c.state.Fstype == "xfs" {
```

Comment:
Do we have any XFS tests in the `tests` repo to exercise this? If not, maybe you could add a small unit test?

Reply:
How can I add a small unit test for that ? I would have to test `createContainer()`, which is a very hard one...

Reply:
I was thinking you could create an invalid `pkg.oci.config` annotation. Then, when you call `createContainer(pod *Pod, c *Container)`, the `json.Unmarshal()` will fail. But `c.state.Fstype`  should then be `"xfs"`.

Reply:
Looking again, I agree - you can't test for this scenario as even if you can force the error, the root mount options aren't returned.